### PR TITLE
IEN-422 first version of apis for reports 3 to 7

### PR DESCRIPTION
**Pull request by bhumin-fw**
- Added API for reports 3 to 7
- APIs with fixed formate and partially confirmed Queries data

### CODE REVIEW

File: apps/api/src/report/report.controller.ts [hunk 1]
```diff
@@ -47,4 +46,81 @@ export class ReportController {
...
+    @Query('from') from: string,
+    @Query('to') to: string,
+  ): Promise<object[]> {
+    this.logger.log(
+      `Report: Applicants count of Active, Withdrawn and Hired (from ${from} to ${to})`,
+    );
+    return this.reportService.getHiredWithdrawnActiveCount(from, to);
+  }
+
+  @ApiOperation({ summary: 'Report 4: hired-withdrawn-active applicants' })
```
Comment (coryatwater-FW) [inline apps/api/src/report/report.controller.ts:67]: summary should be licensing stage applicants?

**Review by coryatwater-FW — APPROVED** [review]
looks good to me

File: apps/api/src/report/report.util.service.ts [hunk 2]
```diff
@@ -119,6 +119,297 @@ export class ReportUtilService {
...
+          t1.id,
+          COALESCE((SELECT max(sa.status_id) FROM public.ien_applicant_status_audit sa 
+            WHERE sa.applicant_id=t1.id AND sa.status_id IN (19, 16)), 0) AS has_license
+        FROM (
+          SELECT 
+            applicants.id,
+            CASE WHEN COALESCE(ien_staus_hired.status_id, 0) > 0 THEN 1 ELSE 0 END as hired,
+            CASE WHEN COALESCE(ien_staus_withdrew.status_id, 0) > 0 THEN 1 ELSE 0 END as withdrawn
+          FROM public.ien_applicants as applicants
+          LEFT JOIN public.ien_applicant_status_audit ien_staus_hired
```
Comment (sunghwan-park-fw) [inline apps/api/src/report/report.util.service.ts:184]: Isn't it same one join with two conditions combined by 'or'?

Comment (bhumin-fw) [reply apps/api/src/report/report.util.service.ts:184]: @sunghwan-park-fw
No, its not same as you may receive both status in some case. In such case we have to run group by and give priority to withdrew first in selection.

Use 2 instance of same table to find each detail separately help us to keep only one raw for an applicant. And we do not need to run a process of group by, partitions and provide row number based on order withdrawal then hired and pic row 0 for the same.

**Review by sunghwan-park-fw — APPROVED** [review]

**Pull request merged**